File: go/dc_06_lets_review.go
package main
import "fmt"

func main() {
    var (
        T, length, i, j int
        text, even, odd string
        isEven bool
        result []string
    )

    //Enter your code here. Read input from STDIN. Print output to STDOUT
    fmt.Scan(&T)

    if (T < 1) || (T > 10) {
        fmt.Println("Not valide number of test cases")
        return
    }

    for i = 0; i < T; i++ {
        fmt.Scan(&text)
        length = len(text)

        if (length < 2) || (length > 10000) {
            fmt.Println("Not valide length of text")
            return
        }

        isEven = true
        even = ""
        odd = ""

        for j = 0; j < length; j++ {
            if isEven {
                isEven = false
                even = even + string(text[j])
            } else {
                isEven = true
                odd = odd + string(text[j])
            }
        }

        result = append(result, fmt.Sprintf("%s %s", even, odd))
    }

    length = len(result)
    for i = 0; i < length; i++ {
        fmt.Println(result[i])
    }
}

File: php/dc_06_lets_review.php
<?php
$_fp = fopen("php://stdin", "r");

/* Enter your code here. Read input from STDIN. Print output to STDOUT */

$countStrings = fgets($_fp);

$output = [];
for ($i = 1; $i <= $countStrings; $i++) {
    $string = trim(fgets($_fp));
    
    $processed = [];
    for ($j = 1; $j <= strlen($string) ; $j++) {
        $index = ($j % 2) ? 'odd' : 'even';
        $processed[$index] = $processed[$index] . $string[$j - 1];
    }
    
    $output[] = sprintf('%s %s', $processed['odd'], $processed['even']);
}

foreach ($output as $string) {
    printf("$string\n");
}

fclose($_fp);
?>
